housekeeper/store/api/sqlalchemy_classes.py

```diff
@@ -0,0 +1,361 @@
+        tags: list = [],
+    ) -> File:
+        new_file = File(
+            version=version,
+            path=path,
+            to_archive=to_archive,
+            tags=[
+                self.get_tag(session=session, name=name)
+                or self.add_tag(session=session, name=name)
+                for name in tags
+            ],
+        )
+        session.add(new_file)
+        return new_file
+
+    def add_tag(self, session, name: str, category: str = None) -> Tag:
+        new_tag = Tag(category=category, name=name)
+        session.add(new_tag)
+        return new_tag
+
+    def get_bundle(self, session, name: str) -> Bundle:
+        query = session.query(Bundle).filter(Bundle.name.like(name))
+        return query.first()
+
+    def get_version(self, session, version_id: int) -> Version:
+        query = session.query(Version).filter(Version.id == version_id)
+        return query.first()
+
+    def get_tag(self, session, name: str) -> Tag:
+        query = session.query(Tag).filter(Tag.name.ilike(name))
+        return query.first()
+
+    def get_file(self, session, file_id: int) -> File:
+        query = session.query(File).filter(File.id == file_id)
+        return query.first()
+
+    def include_file(self, session, file_obj: File, include_path: Path) -> None:
+        from_path = Path(file_obj.path)
+        to_path = Path(include_path, from_path.name)
+        try:
+            from_path.link_to(to_path)
+            file_obj.path = to_path.as_posix()
+        except:
+            # Raise custom error
+            raise
+
+    def include_version(self, session, version_obj: Version) -> None:
+        try:
+            include_path = Path(
+                self.root,
+                version_obj.bundle,
+                version_obj.id,
+                version_obj.created_at.date(),
+            )
+            Path.mkdir(include_path, parents=True, exist_ok=True)
+            for file_obj in version_obj.files:
+                self.include_file(
+                    session=session, file_obj=file_obj, include_path=include_path,
+                )
+            version_obj.include = True
+        except:
+            Path.rmdir(include_path)
+            raise
+
+
+class SessionWrapper:
+    def __init__(self, uri: str, root: Path):
+        self.engine = create_engine(uri, echo=False, poolclass=NullPool)
+        self.root = root
+        super().__init__()
+
+    @contextmanager
+    def session_scope(self):
+        Session.configure(bind=self.engine)
+        session = Session()
+        try:
+            yield session
+            session.commit()
+        except:
+            session.rollback()
+            raise
+        finally:
+            session.close()
+
+
+class ActionHandler(SessionWrapper, BaseActionHandler):
+    """Class that handles logic to be executed within distinct session scope
+    No objects are to be passed as arguments to these methods, since those objects """
```

Comment:
Since those objects ...

😄 looks like that last part of the sentence is missing